lib/context/materials.rb

```diff
@@ -88,12 +88,19 @@ def new_commit(filename, commit, author = 'gouser')
     end
 
     def create_new_directory_and_add_file(filename,directory_name)
-      cd(@path.to_s) do
-
-        mkdir_p(directory_name)
-         cd(directory_name) do
+      file_content=""
```

Comment:
Instead of doing read and write of file wouldnt it be better to just copy the file from `resources` folder to the repo folder if a file with that name exists in `resources` folder? else just a touch of new file with the name